ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/RewrittenAnalysis.java

```diff
@@ -139,18 +139,20 @@ public Optional<WindowExpression> getWindowExpression() {
     final Optional<WindowExpression> windowExpression = original.getWindowExpression();
     final Optional<RefinementInfo> refinementInfo = original.getRefinementInfo();
 
-    /* Return the original window expression, unless there is no grace period provided during a
-    suppression, in which case we rewrite the window expression to have a default grace period
-    of zero.
-    */
-    if (!(windowExpression.isPresent()
-        && !windowExpression.get().getKsqlWindowExpression().getGracePeriod().isPresent()
-        && refinementInfo.isPresent()
-        && refinementInfo.get().getOutputRefinement() == OutputRefinement.FINAL
-         )
-    ) {
+    // we only need to rewrite if we have a window expression and if we use emit final
+    if (!windowExpression.isPresent()
+        || !refinementInfo.isPresent()
+        || !refinementInfo.get().isFinal()) { // can we test for isFinal? or memory vs rocks?
```

Comment:
What's the concern here? Are you saying that users of in-memory stores wouldn't want to enable the new emit strategy? (Probably shouldn't affect ksql since ksql always uses rocks, just trying to understand.)

Reply:
We test for `!isFinal()` that returns `!true` for `FINAL (in-memory) || FINAL_PERSISTENT (rocks)`?

The question is, if we need to test for only `!FINAL_PERSISTENT` instead?

Given you other comment about `FINAL` might be obsolete, the answer is "no". However, if we should return the unmodified window expression only for `FINAL` (but not for `FINAL_PERSISTENT`) the answer might be "yes".

Reply:
Got it, I initially misunderstood what you meant by "memory vs rocks". I would hope that only new queries go through the RewrittenAnalysis so the two should be equivalent. It's been a while since I've worked with this part of the code, though.